Orange/widgets/visualize/owradviz.py

```diff
@@ -360,19 +346,18 @@ def set_data(self, data):
         self._vizrank_color_change()
 
     def check_data(self):
+        def error(err):
+            err()
+            self.data = None
+
         super().check_data()
         if self.data is not None:
             domain = self.data.domain
             if len(self.data) < 2:
-                self.Error.no_instances()
-                self.data = None
+                error(self.Error.no_instances)
             elif len([v for v in domain.variables + domain.metas
                       if v.is_primitive()]) < 3:
-                self.Error.no_features()
-                self.data = None
-            elif np.all(np.isnan(self.data.X).any(axis=1)):
-                self.Warning.missing_coords()
-                self.data = None
+                error(self.Error.no_features)
```

Comment:
Doesn't this condition already ensure that `len(self.model_other) + len(self.model_selected) > 3`? (see `init_attr_values`)

Reply:
No. The error is shown when len(self.model_other) + len(self.model_selected) < 3. So when the number of variables is 3, widget still shows the graph, only vizrank is disabled.